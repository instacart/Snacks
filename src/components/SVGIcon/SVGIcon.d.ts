import * as React from 'react'
import * as icons from './icons'
import { RadiumStyles, Omit } from '../..'

export interface SVGIconProps extends Omit<React.ComponentPropsWithoutRef<'svg'>, 'style'> {
  color?: string
  name: keyof typeof icons
  size?: 'small' | 'standard' | 'large'
  style?: RadiumStyles
}

declare const SVGIcon: React.ComponentClass<SVGIconProps>

export default SVGIcon
