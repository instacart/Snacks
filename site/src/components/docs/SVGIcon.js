import React, {Fragment} from 'react'
import Playground from '../Playground'

export default () => (

SVGIcon example:

<Playground>
{`
  <div>
    <SVGIcon name='cart' size='large' />
    <SVGIcon name='cart' />
    <SVGIcon name='cart' size='small' />
    <SVGIcon
      name='likeFilled'
      color='#fff'
      style={{
        backgroundColor: '#43B02A',
        borderRadius: '50%',
        padding: '5px'
      }}
    />
  </div>
`}
</Playground>

See full list of available icons under the SVGIcons section.

)
