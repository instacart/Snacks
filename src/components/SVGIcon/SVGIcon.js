import React from 'react'
import Radium from 'radium'
import PropTypes from 'prop-types'
import * as icons from './icons'

const sizes = { small: '18px', standard: '24px', large: '36px' }

SVGIcon.propTypes = {
  color: PropTypes.string,
  name: PropTypes.oneOf(Object.keys(icons)).isRequired,
  size: PropTypes.oneOf(Object.keys(sizes)),
}

SVGIcon.defaultProps = {
  color: 'currentColor',
  size: 'standard',
}

function SVGIcon({ color, name, size, ...props }) {
  const Component = icons[name]
  return <Component aria-hidden fill={color} width={sizes[size]} height={sizes[size]} {...props} />
}

export default Radium(SVGIcon)
