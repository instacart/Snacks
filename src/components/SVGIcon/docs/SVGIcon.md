SVGIcon example:

```jsx static
<div>
  <SVGIcon name="cart" size="large" />
  <SVGIcon name="cart" />
  <SVGIcon name="cart" size="small" />
  <SVGIcon
    name="likeFilled"
    color="#fff"
    style={{
      backgroundColor: '#43B02A',
      borderRadius: '50%',
      padding: '5px',
    }}
  />
</div>
```

See full list of available icons under the SVGIcons section.
