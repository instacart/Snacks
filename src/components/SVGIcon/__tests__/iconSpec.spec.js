import React from 'react'
import renderer from 'react-test-renderer'
import SVGIcon from '../SVGIcon'

it('renders accessibility icon correctly', () => {
  const tree = renderer.create(<SVGIcon name="accessibility" />).toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders accessibility icon with styles', () => {
  const tree = renderer.create(<SVGIcon name="accessibility" style={{ fill: '#fff' }} />).toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders accessibility icon with size', () => {
  const tree = renderer.create(<SVGIcon name="accessibility" size="small" />).toJSON()
  expect(tree).toMatchSnapshot()
})
