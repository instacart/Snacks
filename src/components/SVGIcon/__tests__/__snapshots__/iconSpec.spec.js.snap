// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders accessibility icon correctly 1`] = `
<test-file-stub
  aria-hidden={true}
  fill={undefined}
  height="24px"
  width="24px"
/>
`;

exports[`renders accessibility icon with size 1`] = `
<test-file-stub
  aria-hidden={true}
  fill={undefined}
  height="18px"
  width="18px"
/>
`;

exports[`renders accessibility icon with styles 1`] = `
<test-file-stub
  aria-hidden={true}
  fill={undefined}
  height="24px"
  style={
    Object {
      "fill": "#fff",
    }
  }
  width="24px"
/>
`;
